tests/main/user-data-handling-homedirs/task.yaml

```diff
@@ -0,0 +1,72 @@
+summary: Test support for snap refresh in non-standard home directory paths
```

Comment:
Perhaps you could extend the snapd-homedirs test which is pretty similar to this new one. I think you just need to add the second part of the test.